pilot/pkg/status/reporter.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"k8s.io/client-go/rest"
+
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/utils/clock"
+
+	"istio.io/istio/pilot/pkg/model"
+)
+
+func NewIstioContext(stop <-chan struct{}) context.Context {
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		<-stop
+		cancel()
+	}()
+	return ctx
+}
+
+type Reporter struct {
+	mu                  sync.RWMutex
+	status              map[string]string
+	reverseStatus       map[string][]string
+	dirty               bool
+	inProgressResources []Resource
+	client              v1.ConfigMapInterface
+	cm                  *corev1.ConfigMap
+	UpdateInterval      time.Duration
+	PodName             string
+	clock               clock.Clock
+	store               model.ConfigStore
+}
+
+// Starts the reporter, which watches dataplane ack's and resource changes so that it can update status leader
+// with distribution information
+func (r *Reporter) Start(restConfig *rest.Config, namespace string, stop <-chan struct{}) {
+	ctx := NewIstioContext(stop)
+	t := r.clock.Tick(r.UpdateInterval)
+	clientSet, err := kubernetes.NewForConfig(restConfig)
```

Comment:
Why do we need a new client here instead of passing the existing client? as far as I understand client-go best practices are to share the client

Reply:
Where is the existing clientset?